scripts/crc-deploy.sh

```diff
@@ -0,0 +1,193 @@
+  IMAGE_REGISTRY_PORT=5000
+
+  # Create port-forward to CRC registry
+  kubectl port-forward -n ${OPENSHIFT_REGISTRY_NAMESPACE} svc/${IMAGE_REGISTRY_SVC} ${IMAGE_REGISTRY_PORT}:${IMAGE_REGISTRY_PORT} > /dev/null 2>&1 &
+  PORT_FWD_PID=$!
+
+  # Remember to close the port-forward
+  trap 'kill "${PORT_FWD_PID}"' EXIT
+
+  # give port-forward a second
+  # I found that without this docker login would not work
+  # as if it couldn't reach the docker server
+  sleep 1
+
+  # Login to the CRC registry
+  oc whoami -t | docker login localhost:5000 --username user --password-stdin
+
+  # Tag and push olm image
+  echo "Pushing olm image"
+  docker tag "${LOCAL_OLM_IMAGE}" "${CRC_OLM_IMAGE}"
+  docker push "${CRC_OLM_IMAGE}"
+
+  # Tag and push registry image
+  echo "Pushing registry image"
+  docker tag "${LOCAL_OPM_IMAGE}" "${CRC_OPM_IMAGE}"
+  docker push "${CRC_OPM_IMAGE}"
+
+  # Create image streams
+  echo "Creating image streams: ${CRC_OLM_IMAGE} ${CRC_OPM_IMAGE}"
+  OLM_OPENSHIFT=image-registry.openshift-image-registry.svc:5000/openshift/olm:test
+  OPM_OPENSHIFT=image-registry.openshift-image-registry.svc:5000/openshift/opm:test
+  oc import-image olm --from="${OLM_OPENSHIFT}" --confirm
+  oc import-image opm --from="${OPM_OPENSHIFT}" --confirm
+}
+
+# make_manifest_patches takes in two parameters
+# OLM_IMG: the internal registry istag for the olm image
+# OPM_IMG: the internal registry istag for the opm image
+# it creates a helm values files and other manifest patch files applied by
+# scripts/generate_crds_manifests.sh
+function make_manifest_patches {
+  OLM_IMG=${1?Error:\ olm image undefined}
+  OPM_IMG=${2?Error:\ opm image undefined}
+  SED_OPTS=(-e 's#OLM_IMAGE#'"${OLM_IMG}"'#g' -e 's#OPM_IMAGE#'"${OPM_IMG}"'#g')
+
+  # helm values file
+  VALUES_TEMPLATE=$(mktemp)
+  cat << EOF > "${VALUES_TEMPLATE}"
+olm:
+  image:
+    ref: OLM_IMAGE
+catalog:
+  commandArgs: --configmapServerImage=OPM_IMAGE
+  opmImageArgs: --opmImage=OPM_IMAGE
+  image:
+    ref: OLM_IMAGE
+package:
+  image:
+    ref: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${VALUES_TEMPLATE}" > "${CRC_E2E_VALUES}"
+
+  # psm operator patch
+  PSM_OPERATOR_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${PSM_OPERATOR_PATCH_TEMPLATE}"
+- command: update
+  path: spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+- command: update
+  path: spec.template.spec.containers[0].env[1].value
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${PSM_OPERATOR_PATCH_TEMPLATE}" > scripts/psm-operator-deployment.crc.e2e.patch.yaml
+
+  # collect-profiles patch
+  COLLECT_PROFILE_PATCH_TEMPLATE=$(mktemp)
+  cat << EOF > "${COLLECT_PROFILE_PATCH_TEMPLATE}"
+- command: update
+  path: spec.jobTemplate.spec.template.spec.containers[0].image
+  value: OLM_IMAGE
+EOF
+
+  sed "${SED_OPTS[@]}" "${COLLECT_PROFILE_PATCH_TEMPLATE}" > scripts/collect-profiles.crc.e2e.patch.yaml
+}
+
+# YQ for applying yaml patches
+YQ="go run ./vendor/github.com/mikefarah/yq/v3/"
```

Comment:
See my earlier comment about running into issues when getting to the part in this workflow where we use yq. I tried running the following command locally thinking we might have support for overriding YQ:

```console
$ make crc-deploy YQ=/usr/local/bin/yq
...
go run: no packages loaded from ./vendor/github.com/mikefarah/yq/v3/
```

Maybe this is something we can override just like the KUBECONFIG environment variable?

Reply:
Andddd I tried manually overriding this in the crc-deploy.sh bash script and realized I have a v4 version locally:

```console
$ make crc-deploy YQ=/usr/local/bin/yq
...
Error: write inplace cannot be used with split file
make: *** [Makefile:180: crc-deploy] Error 1
$ yq --version
yq (https://github.com/mikefarah/yq/) version 4.23.1
```

Reply:
It looks like we're stranding some changes made while generating the CRC patches:

```console
$ git status
	modified:   manifests/0000_50_olm_07-olm-operator.deployment.ibm-cloud-managed.yaml
	modified:   manifests/0000_50_olm_07-olm-operator.deployment.yaml
	modified:   manifests/0000_50_olm_08-catalog-operator.deployment.ibm-cloud-managed.yaml
	modified:   manifests/0000_50_olm_08-catalog-operator.deployment.yaml
	modified:   pkg/manifests/csv.yaml
```

Maybe we need to wrap this in it's own function so we can enact a cleanup function (e.g. `git restore manifests`) when we run into a non-zero exit code?